kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/io/kotest/assertions/nondeterministic/ContinuallyTest.kt

```diff
@@ -31,54 +44,103 @@ class ContinuallyTest : FunSpec() {
             intervalFn = DurationFn { 300.milliseconds }
          }
          var k = 0
-         continually(config) {
-            (System.currentTimeMillis() > 0) shouldBe true
+         val result = testContinually(config) {
+            1 shouldBe 1
             k++
          }
          k shouldBe 3
+         result.invocationTimes shouldHaveSize 3
+         result.invocationTimes.shouldForAll { it <= config.duration }
       }
 
-      test("invoke the listener for each successfull inovcation") {
+      test("invoke the listener for each successful invocation") {
          var listened = 0
          var invoked = 0
          val config = continuallyConfig<Unit> {
             duration = 500.milliseconds
             listener = { _, _ -> listened++ }
          }
-         continually(config) {
-            (System.currentTimeMillis() > 0) shouldBe true
+         val result = testContinually(config) {
+            1 shouldBe 1
             invoked++
          }
          invoked shouldBe listened
+         result.invocationTimes shouldHaveSize listened
       }
 
       test("fail broken tests immediately") {
-         shouldThrowAny {
-            continually(12.hours) {
+         val start = testCoroutineScheduler.timeSource.markNow()
+         val failure = shouldFail {
+            testContinually(1.minutes) {
                false shouldBe true
             }
          }
+         failure.shouldHaveMessage("expected:<true> but was:<false>")
+         start.elapsedNow() shouldBe Duration.ZERO
       }
 
       test("fail should throw the underlying error") {
-         shouldThrowExactly<AssertionError> {
-            continually(12.hours) {
+         val start = testCoroutineScheduler.timeSource.markNow()
+         shouldFail {
+            testContinually(1.minutes) {
                throw AssertionError("boom")
             }
          }.message shouldBe "boom"
+         start.elapsedNow() shouldBe Duration.ZERO
       }
 
       test("fail tests start off as passing then fail within the period") {
          var n = 0
-         val e = shouldThrow<Throwable> {
-            continually(3.seconds) {
+         val failure = shouldFail {
+            testContinually(3.seconds) {
                delay(10)
                (n++ < 10) shouldBe true
             }
          }
-         val r =
-            "Test failed after \\d+ms; expected to pass for 3000ms; attempted 100 times\nUnderlying failure was: 100 should be < 100".toRegex()
-         e.message?.matches(r) ?: (false shouldBe true)
+         n shouldBe 11
+         failure shouldHaveMessage "Test failed after 360ms; expected to pass for 3s; attempted 10 times\nUnderlying failure was: expected:<true> but was:<false>"
```

Comment:
Minor note: previously the regex assertion wasn't triggered, so I changed the expected message from `Underlying failure was: 100 should be < 100` to `Underlying failure was: 10 should be < 10`.